src/coreclr/inc/corinfoinstructionset.h

```diff
@@ -128,41 +128,89 @@ enum CORINFO_InstructionSet
 struct CORINFO_InstructionSetFlags
 {
 private:
-    uint64_t _flags = 0;
+    static const int32_t FlagsFieldCount = 1;
+    static const int32_t BitsPerFlagsField = sizeof(uint64_t) * 8;
+    uint64_t _flags[FlagsFieldCount] = { };
+
+
+    static uint32_t GetFlagsFieldIndex(CORINFO_InstructionSet instructionSet)
+    {
+        uint32_t bitIndex = (uint32_t)instructionSet;
+        return (uint32_t)(bitIndex / (uint32_t)BitsPerFlagsField);
+    }
+
+    static uint64_t GetRelativeBitIndex(CORINFO_InstructionSet instructionSet)
+    {
+        return ((uint64_t)1) << (instructionSet & 0x3F);
```

Comment:
This looks like a bug, since it ignores the top 2 bits

```suggestion
        uint32_t bitIndex = (uint32_t)instructionSet;
        return (uint32_t)(bitIndex % (uint32_t)BitsPerFlagsField);
```

Reply:
`GetFlagsFieldIndex` gets the 64-bit field in which a given `instructionSet` is tagged (only the bits 6 and higher matter).

`GetRelativeBitIndex` then gets the bit offset within that field, which can only be `0-63`, so `0x3F` should be the correct mask.

Reply:
`GetRelativeBitMask` might have been a better name, since its not getting the index but rather a mask that only includes the given index.

Reply:
Have changed name to `GetRelativeBitMask `